sale_generator/models/sale_order.py

```diff
@@ -0,0 +1,50 @@
+#  @author Sébastien BEAU <sebastien.beau@akretion.com>
+#  @author EBII MonsieurB <monsieurb@saaslys.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import fields, models, api, _
+from lxml import etree
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    generator_id = fields.Many2one(
+        comodel_name="sale.generator", string="Generator"
+    )
+    is_template = fields.Boolean(string="Is a generator template")
+    active = fields.Boolean("Active", default=True)
+
+    @api.model
+    def create(self, vals):
+        if self.env.context.get("tmpl_mode"):
+            vals["name"] = self.env["ir.sequence"].next_by_code(
+                "sale.order.tmpl"
+            ) or _("New template")
+            if not vals.get("partner_id"):
+                # in tmpl_mode we hide partner_id and don't care about its value
+                vals["partner_id"] = (
+                    self.env["res.partner"].search([], limit=1).id
+                )
+        return super().create(vals)
+
+    @api.model
+    def fields_view_get(
```

Comment:
I do not understand why you did this.
Not after testing If I go to the sale template I need to fill a customer the field should be hidden

Reply:
maybe better to reuse view https://github.com/OCA/sale-workflow/commit/0c384f48d465eea9fa3f0f73094f4900eca6426f#diff-ab867746dba3d99c7ab6259ae0861977